registration/scripts/directives.js

```diff
@@ -56,7 +63,13 @@
                             if (angular.isDefined(deferred)) {
                                 deferred.notify('reject', response);
                             }
+                            apiError = true;
                         });
+                    } else {
+                        //check there are no other error messages on the screen from the api
+                        if (!apiError) {
+                            scope.emptyEmail = true;
+                        }
```

Comment:
`emptyEmail` feels too specific. Instead could it be that the email is required, and when not filled in it's validation will fail and so create field errors? I think it should be handled that way, otherwise what if you need to mark other fields as required and error when they're empty?